pypeit/core/wavecal/wvutils.py

```diff
@@ -31,6 +31,38 @@ def parse_param(par, key, slit):
 
     return param
 
+
+def get_delta_wave(wave, gpm):
+    """
+    Given an input wavelength vector and an input good pixel mask, this code computes the delta_wave defined
+    to be wave_old[i]-wave_old[i-1]. The missing point at the end of the array is just appended to have an
+    equal size array.
+
+    Args:
+        wave (float ndarray): shape = (nspec,)
+            Array of input wavelengths. Muse be a one dimensional array.
+        gpm (bool ndarray): shape = (nspec)
+            Boolean good pixel mask defining where the wave_old are good.
+
+    Returns
+    -------
+    delta_wave (float ndarray): shape = (nspec,)
+            Array of wavelength differences using np.diff, where the last pixel has been extrapolated
+
+    """
+
+    delta_wave = np.zeros_like(wave)
+    wave_diff = np.diff(wave[gpm])
+    wave_diff = np.append(wave_diff, wave_diff[-1])
+    delta_wave[gpm] = wave_diff
```

Comment:
Is this intended? The `delta_wave` is 0 for a bad pixel. Also, `delta_wave` for pixels to the left of a bad pixel will have a spuriously large `delta_wave` (because the diff is done only on the good pixels, on line 55).

Reply:
The gpm for wavelengths is wave > 1. There are thus never any bad pixels, just regions where the wavelengths are not defined because of echelle orders that end (i.e. the spec_min, spec_max) functionality. This is not an elegant solution, but there was no other way to store variable length wavelength and flux arrays in numpy arrays of fixed shape. So in multiple places we define a good wavelength mask via wave > 1, where  wave = 0 are the bad values. Thus, the way that this is defined it will always give the right dlambda.